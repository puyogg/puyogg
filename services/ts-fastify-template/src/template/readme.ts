import * as fs from 'node:fs/promises';
import * as path from 'node:path';
import * as url from 'node:url';

const __dirname = path.dirname(url.fileURLToPath(import.meta.url));

export const readme = async (packageName: string): Promise<string> => {
  const baseReadme = await fs.readFile(path.resolve(__dirname, '../../README.md'), {
    encoding: 'utf-8',
  });

  const checklist = baseReadme.split('## New Lib Checklist\n\n')[1].split('\n##')[0];

  const newReadme = `# ${packageName}

TODO: package description

## New Library Checklist

${checklist}`;
  return newReadme;
};
