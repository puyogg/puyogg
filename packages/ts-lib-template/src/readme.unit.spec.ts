import { expect, test } from 'vitest';
import { readme } from './readme.js';

test('creates readme string with checklist and header', async () => {
  const result = await readme('MyPackage');
  expect(result).toMatchSnapshot();
});
