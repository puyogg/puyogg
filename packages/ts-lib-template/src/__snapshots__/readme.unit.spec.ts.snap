// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`creates readme string with checklist and header 1`] = `
"# MyPackage

TODO: package description

## New Library Checklist

- [ ] Update README.md
- [ ] Update TypeScript, ESLint, Prettier, and their configs (if necessary)
- [ ] Add project references to \`tsconfig.json\`
- Configure these package.json scripts that CI/CD looks for, if your package needs them:
  - [ ] \`typecheck\`
  - [ ] \`build\`
  - [ ] \`test:unit\`
  - [ ] \`test:int\` (Integration tests)
  - [ ] \`build-image\` (Docker image)
"
`;
